### PR TITLE
[#551] Notify on Real Changes only

### PR DESCRIPTION
**Pull request by cdietrich**
[#551] Notify on Real Changes only
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java [hunk 1]
```diff
@@ -528,11 +528,13 @@ protected XtextResource getState() {
...
 					try {
-						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) { 
+						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) {
+							boolean wasHadUpdates = hadUpdates;
 							hadUpdates = false;
 							if (getCancelIndicator().isCanceled() && isCancelable)
 								throw new OperationCanceledException();
-							notifyModelListeners(state);
+							if (wasHadUpdates)
+								notifyModelListeners(state);
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java]: code style: shall we encapsulate single-line statements in blocks?

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java [hunk 2]
```diff
@@ -528,11 +528,13 @@ protected XtextResource getState() {
 					throw e.getWrapped();
 				} finally {
 					try {
-						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) { 
+						if(potentialUpdaterCount.decrementAndGet() == 0 && (hadUpdates || isCancelReaders)) {
+							boolean wasHadUpdates = hadUpdates;
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java]: strange variable name.
how about setting hadUpdates in a finally-block to false after notifying the listeners? I did not see that it is mandatory to set it to false before.

```
try {
   ...
   if (hasUpdates) { notifyModelListeners(state); }
} finally {
   hadUpdates = false;
}
```

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocument.java]: i just kept the existing styles ;)

**Comment by cdietrich** [comment]
ping @szarnekow

**Pull request merged**